modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/InjectedFieldsModelValidator.java

```diff
@@ -92,8 +96,10 @@ protected void onOperation(HasOperationModels owner, OperationModel model) {
 
           @Override
           protected void onConnectionProvider(HasConnectionProviderModels owner, ConnectionProviderModel model) {
-            validateFields(model, getImplementingType(model), DefaultEncoding.class);
-            validateFields(model, getImplementingType(model), RefName.class);
+            Optional<Class> implementingType = getImplementingType(model);
+            validateFields(model, implementingType, DefaultEncoding.class, String.class);
```

Comment:
instead of calling this method 3 times, shouldn't it be refactored so that all the annotations and types are passed together and the list of fields is iterated only once?

Reply:
can you explain this change? this doesn't provide the proposed optimization

Reply:
I will have to pass the `ConnectionProviderModel`, the `implementingType`, and a list or map with the types and annotations. It looks quite ugly to me. Is there any benefit between calling three times a  method and a foreach that requires creating the list and map?

On the other hand it's not exactly the same of parameters, so I'll end up with code duplication anyway